### PR TITLE
Bug fixes: planar images in halide_buffer_display, way to add input i…

### PR DESCRIPTION
**Pull request by steven-johnson**
…mages to bundle.  Added timing macro

### CODE REVIEW

File: apps/templates/SimpleAppAPI.h [hunk 1]
```diff
@@ -19,4 +19,61 @@ int halide_buffer_display(const buffer_t* buffer);
...
+
+// In a program that calls the halide function repeatedly, e.g. once per frame
+// set the number of iterations to compute statistics over.
+#define PROFILE_ITERATIONS 1
+#ifdef PROFILE_SHADER
+#import <mach/mach_time.h>
+
+#define PROFILE_BEGIN(msg) \
+{\
+  uint64_t begin_time = mach_absolute_time();
```
Comment (steven-johnson) [inline apps/templates/SimpleAppAPI.h]: If we use halide_current_time_ns() here, we could dispense with the ifdef APPLE stuff

Comment (abestephensg) [inline apps/templates/SimpleAppAPI.h]: That's a good idea.

File: apps/templates/SimpleAppAPI.h [hunk 2]
```diff
@@ -19,4 +19,61 @@ int halide_buffer_display(const buffer_t* buffer);
...
+  minDuration = (duration < minDuration) ? duration : minDuration;\
+  maxDuration = (duration > maxDuration) ? duration : maxDuration;\
+\
+  if (!(++iterations % PROFILE_ITERATIONS)) {\
+\
+    const size_t pixelsPerFrame = (w) * (h);\
+    float Mpixelpers = (((float)pixelsPerFrame*(float)PROFILE_ITERATIONS)/1e6) / totalDuration;\
+    float avgDuration = totalDuration / (float)PROFILE_ITERATIONS;\
+\
+    NSLog(@"H>> %s %3.3f Mpixels/sec %3.3f avg sec %3.3f min %3.3f max (%d iterations)\n", \
```
Comment (abestephensg) [inline apps/templates/SimpleAppAPI.h]: If we are going to make this platform independent, we need a substitute for NSLog here as well.

Comment (steven-johnson) [inline apps/templates/SimpleAppAPI.h]: > vararg version of halide_print for this harness

make it so

**Pull request merged**